Upstream: yes

From 5927885f7673cfa50854687c34f50da13435fb93 Mon Sep 17 00:00:00 2001
From: doko <doko@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 7 Sep 2017 07:20:51 +0000
Subject: [PATCH] 2017-09-07  Matthias Klose  <doko@ubuntu.com>

        * include/x86_64-signal.h (HANDLE_DIVIDE_OVERFLOW): Replace
        'struct ucontext' with ucontext_t.
        * include/i386-signal.h (HANDLE_DIVIDE_OVERFLOW): Likewise.
        * include/s390-signal.h (HANDLE_DIVIDE_OVERFLOW): Likewise.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-6-branch@251831 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libjava/ChangeLog               | 7 +++++++
 libjava/include/i386-signal.h   | 2 +-
 libjava/include/s390-signal.h   | 2 +-
 libjava/include/x86_64-signal.h | 2 +-
 4 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/libjava/ChangeLog b/libjava/ChangeLog
index 4e62c3666fe..15e89178440 100644
--- a/libjava/ChangeLog
+++ b/libjava/ChangeLog
@@ -1,3 +1,10 @@
+2017-09-07  Matthias Klose  <doko@ubuntu.com>
+
+	* include/x86_64-signal.h (HANDLE_DIVIDE_OVERFLOW): Replace
+	'struct ucontext' with ucontext_t.
+	* include/i386-signal.h (HANDLE_DIVIDE_OVERFLOW): Likewise.
+	* include/s390-signal.h (HANDLE_DIVIDE_OVERFLOW): Likewise.
+
 2017-07-04  Release Manager
 
 	* GCC 6.4.0 released.
diff --git a/libjava/include/i386-signal.h b/libjava/include/i386-signal.h
index c2409b0e301..ef77e7e99ae 100644
--- a/libjava/include/i386-signal.h
+++ b/libjava/include/i386-signal.h
@@ -29,7 +29,7 @@ static void _Jv_##_name (int, siginfo_t *,			\
 #define HANDLE_DIVIDE_OVERFLOW						\
 do									\
 {									\
-  struct ucontext *_uc = (struct ucontext *)_p;				\
+  ucontext_t *_uc = (ucontext_t *)_p;					\
   gregset_t &_gregs = _uc->uc_mcontext.gregs;				\
   unsigned char *_eip = (unsigned char *)_gregs[REG_EIP];		\
 									\
diff --git a/libjava/include/s390-signal.h b/libjava/include/s390-signal.h
index 4ca4c108864..9261b52be6e 100644
--- a/libjava/include/s390-signal.h
+++ b/libjava/include/s390-signal.h
@@ -51,7 +51,7 @@ do									\
   struct                                                                \
   {                                                                     \
     unsigned long int uc_flags;                                         \
-    struct ucontext *uc_link;                                           \
+    ucontext_t *uc_link;                                                \
     stack_t uc_stack;                                                   \
     mcontext_t uc_mcontext;                                             \
     unsigned long sigmask[2];                                           \
diff --git a/libjava/include/x86_64-signal.h b/libjava/include/x86_64-signal.h
index 12383b5485a..e36c5a32a64 100644
--- a/libjava/include/x86_64-signal.h
+++ b/libjava/include/x86_64-signal.h
@@ -28,7 +28,7 @@ static void _Jv_##_name (int, siginfo_t *,			\
 #define HANDLE_DIVIDE_OVERFLOW						\
 do									\
 {									\
-  struct ucontext *_uc = (struct ucontext *)_p;				\
+  ucontext_t *_uc = (ucontext_t *)_p;					\
   gregset_t &_gregs = _uc->uc_mcontext.gregs;				\
   unsigned char *_rip = (unsigned char *)_gregs[REG_RIP];		\
 									\
-- 
2.16.3

